src/protocols/MeterExec.cpp

```diff
@@ -38,44 +43,159 @@ MeterExec::MeterExec(std::list<Option> options)
 	} catch (vz::VZException &e) {
-		print(log_error, "Missing command or invalid type", "");
+		print(log_error, "MeterExec::MeterExec: Missing command or invalid type", name().c_str());
 		throw;
 	}
 
+	// an optional format string for scanf()
 	try {
-		_format = optlist.lookup_string(options, "format");
+		const char *config_format = optlist.lookup_string(options, "format");
+
+		/**
+		 * Compiling the provided format string in a format string for scanf
+		 * by replacing the following tokens
+		 *
+		 * "$v" => "%1$lf" (value)
+		 * "$i" => "%2$ms" (identifier)		(memory gets allocated by sscanf())
+		 * "$t" => "%3$lf" (timestamp)
+		 */
+
+		int config_len = strlen(config_format);
+		int scanf_len = config_len + 15; // adding extra space for longer conversion specification in scanf_format
+
+		char *scanf_format = (char *)malloc(scanf_len); // the scanf format string
+
+		int i = 0; // index in config_format string
+		int j = 0; // index in scanf_format string
+		while (i <= config_len && j <= scanf_len) {
+			switch (config_format[i]) {
+					case '$':
+						if (i+1 < config_len) { // introducing a token
+							switch (config_format[i+1]) {
+									case 'v': j += sprintf(scanf_format+j, "%%1$lf"); break;
+									case 'i': j += sprintf(scanf_format+j, "%%2$ms"); break;
+									case 't': j += sprintf(scanf_format+j, "%%3$lf"); break;
+									// Regex is not working with gcc-4.6
+									// (?<identifier>[\w]+): (?<value>[\d]+[,|.]?[?:\d]+) (?<timestamp>\d{10})
+									//case 'v': j += sprintf(scanf_format+j, "([\\d]+[.|,]?[?:\\d]+)"); break;
+									//case 'i': j += sprintf(scanf_format+j, "([\\w]+)"); break;
+									//case 't': j += sprintf(scanf_format+j, "(\\d{10})"); break;
+							}
+							i++;
+						}
+						break;
+
+					case '%':
+						scanf_format[j++] = '%'; // add double %% to escape a conversion identifier
+						// nobreak;
+					default:
+						scanf_format[j++] = config_format[i]; // just copying
+			}
+
+			i++;
+		}
+
+		print(log_debug, "MeterExec::MeterExec: Parsed format string \"%s\" => \"%s\"", name().c_str(), config_format, scanf_format);
+		_format = scanf_format;
 	} catch (vz::OptionNotFoundException &e) {
-		_format = NULL; /* use default format */
+		_format = ""; // use default format
 	} catch (vz::VZException &e) {
-		print(log_error, "Failed to parse format", "");
+		print(log_error, "MeterExec::MeterExec: Failed to parse format", name().c_str());
 		throw;
 	}
 }
 
 MeterExec::~MeterExec() {
+}
 
-	free((void*)_command);
+int MeterExec::open() {
+	print(log_debug, "MeterExec::open: Testing command line '%s': %s", name().c_str(), command(), strerror(errno));
+	_pipe = popen(command(), "r");
 
-	if (_format != NULL) {
-		free((void*)_format);
+	if (_pipe == NULL) {
+		print(log_error, "MeterExec::open: popen(%s): %s", name().c_str(), command(), strerror(errno));
+		return ERR;
 	}
-}
 
-int MeterExec::open() {
+/* Pipe is closed after script complete
+ 	if (_pipe != NULL) {
```

Comment:
are you sure that the pipe doesn't need to be closed?
Even if the process ends what happens with the file handle? We might run out of file handles on the system if pclose is not performed.